apps/jobs/models.py

```diff
@@ -58,3 +64,57 @@ def __unicode__(self):
     class Meta:
         app_label = 'jobs'
         db_table = 'submission'
+
+    def save(self, *args, **kwargs):
+
+        if not self.pk:
+            sub_num = Submission.objects.filter(
+                challenge_phase=self.challenge_phase,
+                participant_team=self.participant_team).aggregate(
+                Max('submission_number'))['submission_number__max']
+            if sub_num:
+                self.submission_number = sub_num + 1
+            else:
+                self.submission_number = 1
+
+            failed_count = Submission.objects.filter(
+                challenge_phase=self.challenge_phase,
+                participant_team=self.participant_team,
+                status=Submission.FAILED).count()
+
+            offset_submission_count = self.submission_number - failed_count
+
+            if (offset_submission_count > self.challenge_phase.max_submissions):
+                print "Checking to see if the offset_submission_count (%d) ", \
+                    " is greater than the maximum allowed (%d)" % (
+                        offset_submission_count, self.challenge_phase.max_submissions)
+                raise PermissionDenied(
+                    "The maximum number of submissions has been reached.")
+            else:
+                print "Submission number below maximum."
+
+            if hasattr(self.challenge_phase, 'max_submissions_per_day'):
+                submissions_done_today_count = Submission.objects.filter(
+                    challenge_phase__challenge=self.challenge_phase.challenge,
+                    participant_team=self.participant_team,
+                    challenge_phase=self.challenge_phase,
+                    submitted_at__gte=datetime.date.today()).count()
+
+                if ((submissions_done_today_count + 1 - failed_count > self.challenge_phase.max_submissions_per_day) or
+                        (self.challenge_phase.max_submissions_per_day == 0)):
+                    print 'PERMISSION DENIED'
+                    raise PermissionDenied(
```

Comment:
If permission is denied then on save, it will throw an error, right? This I guess will call 500 Internal error. We should catch this exception and throw a 401 error saying that max number of submissions have been made.